### ISSUE TITLE
Type definitions not properly sorted in gen-python

### ISSUE THREAD

**Issue by kervel** [linkml/linkml#1365]
**Describe the bug**

When creating type hierarchies using uriorcurie, in gen-python, classes are emitted in the wrong order causing a non-working generated python module.

**To reproduce**

Generate a python module using gen-python using the following schema:

```yaml
id: http://example.org
name: sth
imports:
  - https://w3id.org/linkml/types
prefixes:
  x: http://example.org/
default_prefix: x
default_range: string
description: test


types:
  category type:
    typeof: uriorcurie
    description: >-
      a category
```

the class definition of `CategoryType` comes before the one of `Uriorcurie`, which causes:

```
vscode ➜ /workspace (feature/signal-links-v3) $ python tst.py
Traceback (most recent call last):
  File "/workspace/tst.py", line 42, in <module>
    class CategoryType(Uriorcurie):
NameError: name 'Uriorcurie' is not defined. Did you mean: 'URIorCURIE'?
```

**Comment by kervel** [comment]
PR: https://github.com/linkml/linkml/pull/1366

**Comment by cmungall** [comment]
fixed by #1366, many thanks!
